python/lsst/summit/utils/utils.py

```diff
@@ -1419,3 +1425,221 @@ def getSunAngle(time: Time | None = None) -> float:
         time = Time.now()
     sun_altaz = get_sun(time).transform_to(AltAz(obstime=time, location=SIMONYI_LOCATION))
     return sun_altaz.alt.deg
+
+
+@dataclass
+class RobustFitResult:
+    intercept: float
+    slope: float
+    scatter: float
+    slope_pvalue: float
+    slope_stderr: float
+    slope_tvalue: float
+    intercept_pvalue: float
+    intercept_stderr: float
+    intercept_tvalue: float
+
+
+class RobustFitter:
+    """
+    Robust linear fit using RANSAC,
+    with confidence and prediction intervals from statsmodels.
+    """
+
+    def __init__(
+        self,
+        x: np.ndarray,
+        y: np.ndarray,
+        min_samples: float = 0.2,
+        residual_threshold: float | None = None,
+        random_state: int = 42,
+    ):
+        self.min_samples = min_samples
+        self.residual_threshold = residual_threshold
+        self.random_state = random_state
+
+        if self.residual_threshold is None:
+            # Set a default residual threshold based on the data
+            if np.isnan(x).all() or np.isnan(y).all():
+                raise ValueError("Cannot fit model: all input data is NaN.")
+            self.residual_threshold = 1.5 * mad_std(y)
+
+        self.fit(x, y)
```

Comment:
why call `self.fit()` on the init? Why not do what we talked about, and make a `RobustFitter` object, and have the fit method return things, so that the main object can be fully reused? There's good reasons for the pattern I keep steering you towards: for example, you have duplicated code validating for not-nan here and in the fit method. Also, you calculate `residual_threshold` here, but not in the fit method, so if someone does try to reuse this object, it will automatically pick up the `residual_threshold` from the previous set of data, so you either need to make them single-use or fully reusable.